src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -0,0 +1,10581 @@
+    __ mv(flag, tmp); // we can use the value of tmp as the result here
+
+    __ j(cont);
+
+    // Handle existing monitor.
+    __ bind(object_has_monitor);
+    // The object's monitor m is unlocked iff m->owner == NULL,
+    // otherwise m->owner may contain a thread or a stack address.
+    //
+    // Try to CAS m->owner from NULL to current thread.
+    __ add(tmp, disp_hdr, (ObjectMonitor::owner_offset_in_bytes() - markWord::monitor_value));
+    __ cmpxchg(/*memory address*/tmp, /*expected value*/zr, /*new value*/xthread, Assembler::int64, Assembler::aq,
+             Assembler::rl, /*result*/flag); // cas succeeds if flag == zr(expected)
+
+    // Store a non-null value into the box to avoid looking like a re-entrant
+    // lock. The fast-path monitor unlock code checks for
+    // markWord::monitor_value so use markWord::unused_mark which has the
+    // relevant bit set, and also matches ObjectSynchronizer::slow_enter.
+    __ mv(tmp, (address)markWord::unused_mark().value());
+    __ sd(tmp, Address(box, BasicLock::displaced_header_offset_in_bytes()));
+
+    __ beqz(flag, cont); // CAS success means locking succeeded
+
+    __ bne(flag, xthread, cont); // Check for recursive locking
+
+    // Recursive lock case
+    __ mv(flag, zr);
+    __ ld(tmp, Address(disp_hdr, ObjectMonitor::recursions_offset_in_bytes() - markWord::monitor_value));
+    __ add(tmp, tmp, 1u);
+    __ sd(tmp, Address(disp_hdr, ObjectMonitor::recursions_offset_in_bytes() - markWord::monitor_value));
+
+    __ bind(cont);
+  %}
+
+  // using cr flag to indicate the fast_unlock result: 0 for success; others failed.
+  enc_class riscv_enc_fast_unlock(iRegP object, iRegP box, iRegPNoSp tmp1, iRegPNoSp tmp2) %{
+    C2_MacroAssembler _masm(&cbuf);
+    Register flag = t1;
+    Register oop = as_Register($object$$reg);
+    Register box = as_Register($box$$reg);
+    Register disp_hdr = as_Register($tmp1$$reg);
+    Register tmp = as_Register($tmp2$$reg);
+    Label cont;
+    Label object_has_monitor;
+
+    assert_different_registers(oop, box, tmp, disp_hdr, flag);
+
+    if (UseBiasedLocking && !UseOptoBiasInlining) {
+      __ biased_locking_exit(oop, tmp, cont, flag);
+    }
+
+    // Find the lock address and load the displaced header from the stack.
+    __ ld(disp_hdr, Address(box, BasicLock::displaced_header_offset_in_bytes()));
+
+    // If the displaced header is 0, we have a recursive unlock.
+    __ mv(flag, disp_hdr);
+    __ beqz(disp_hdr, cont);
+
+    // Handle existing monitor.
+    __ ld(tmp, Address(oop, oopDesc::mark_offset_in_bytes()));
+    __ andi(t0, disp_hdr, markWord::monitor_value);
+    __ bnez(t0, object_has_monitor);
+
+    // Check if it is still a light weight lock, this is true if we
+    // see the stack address of the basicLock in the markWord of the
+    // object.
+
+    __ cmpxchg(/*memory address*/oop, /*expected value*/box, /*new value*/disp_hdr, Assembler::int64, Assembler::relaxed,
```

Comment:
Could you adjust indentation here to align with https://github.com/yadongw/riscv-port-jdk17u/blob/47916a22eed5c92d56a330a481635c355274062f/src/hotspot/cpu/riscv/riscv.ad#L2332-L2336